report/metrics_file_descriptors.go

```diff
@@ -31,29 +37,45 @@ func SetupLinuxBSDFDMetrics(logger *logp.Logger, reg *monitoring.Registry, proce
 }
 
 func FDUsageReporter(logger *logp.Logger, processStats *process.Stats) func(_ monitoring.Mode, V monitoring.Visitor) {
+	p := psprocess.Process{
+		Pid: int32(os.Getpid()),
+	}
```

Comment:
[Question]

If I understand it correctly, right now, those metrics are always from the process itself. However depending on hostFS it might be the PID the container sees or the host.
Therefore I wonder if using `os.Getpid()` might get the wrong PID

Reply:
I don't know, I learnt we have it while working with the beats and k8s/docker.
I know it's used in k8s to avoid giving elevated permissions to a container, so some host paths are mounted on the container, that way the container sees the host but without requiring permissions to do so.
But it's a good question, let me try to dig and see if I find an answer